### PR TITLE
Agent functional tests cleaning and update

### PR DESCRIPTION
**Pull request by julmon**

### CODE REVIEW

**Comment by julmon** [comment]
I'm finally done with this. Please go forward with the code review.

Thanks!

**Comment by bersace** [comment]
Could you rebase fixups to ease review ?

File: temboard-agent/test/test_administration.py [hunk 1]
```diff
@@ -1,15 +1,27 @@
...
+import inspect
 
 from urllib2 import HTTPError
-from test.temboard import init_env, drop_env, rand_string, temboard_request
-import test.configuration as cf
-from test.spc import connector, error
+from test.temboard import init_env, drop_env, temboard_request
+
+# Import spc
+tbda_dir = os.path.realpath(
```
Comment (bersace) [inline temboard-agent/test/test_administration.py]: `tbda_dir = os.path.realpath(os.path.join(__file__, '..', '..', 'temboard-agent')` ?

File: temboard-agent/test/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Temboard Agent functional tests
+
+This directory contains functional tests for each plugin. To run these tests you'll need:
+  * PostgreSQL binaries of the version you want the agent will be connected to when running the tests
+  * `nosetests`
+
+## `nosetests` installation
+
+``` console
+# pip install nosetests
```
Comment (bersace) [inline temboard-agent/test/README.md]: It's `nose` on pypi.

**Comment by bersace** [comment]
I got this error:

```console
(my-venv) root@tb-tests:/usr/local/src/dalibo/temboard/temboard-agent/test# nosetests -v test_api.py
Failure: ImportError (No module named temboardagent.spc) ... ERROR

======================================================================
ERROR: Failure: ImportError (No module named temboardagent.spc)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/src/dalibo/temboard/temboard-agent/my-venv/local/lib/python2.7/site-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/local/src/dalibo/temboard/temboard-agent/my-venv/local/lib/python2.7/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/local/src/dalibo/temboard/temboard-agent/my-venv/local/lib/python2.7/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/usr/local/src/dalibo/temboard/temboard-agent/test/test_api.py", line 8, in <module>
    from test.temboard import init_env, drop_env, rand_string, temboard_request
  File "/usr/local/src/dalibo/temboard/temboard-agent/test/test/temboard.py", line 28, in <module>
    from temboardagent.spc import connector, error  # noqa
ImportError: No module named temboardagent.spc

----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (errors=1)
(my-venv) root@tb-tests:/usr/local/src/dalibo/temboard/temboard-agent/test#
```

**Comment by julmon** [comment]
@bersace maybe we should try with inspect ? :)

**Comment by bersace** [comment]
Nickel :)

``` console
(my-lxc-venv) postgres@tb-tests:/usr/local/src/dalibo/temboard-relecture/temboard-agent/test$ TBD_PGBIN=/usr/lib/postgresql/9.4/bin/ TBD_WORKPATH=/tmp/tbdtests nosetests -v test_api.py
[api] 00: PostgreSQL instance is up & running ... ok
[api] 01: POST /login : HTTP return code is 200 on valid credentials ... ok
[api] 02: POST /login : Session ID format matches ^[a-f0-9]{64}$ ... ok
[api] 03: POST /login : Wrong password must return 404 HTTP code ... ok
[api] 04: POST /login : Wrong username must return 404 HTTP code ... ok
[api] 05: POST /login : Bad username format must return 406 HTTP code ... ok
[api] 06: POST /login : Bad password format must return 406 HTTP code ... ok

----------------------------------------------------------------------
Ran 7 tests in 12.426s

OK
(my-lxc-venv) postgres@tb-tests:/usr/local/src/dalibo/temboard-relecture/temboard-agent/test$
```

**Pull request merged**